app/src/androidTest/java/de/test/antennapod/service/playback/PlaybackServiceTaskManagerTest.java

```diff
@@ -123,6 +129,42 @@ public void update(EventDistributor eventDistributor, Integer arg) {
         pstm.shutdown();
     }
 
+    @Test
+    public void testQueueUpdatedUponDownloadComplete() throws Exception {
+        final Context c = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        { // Setup test data
+            List<FeedItem> queue = writeTestQueue("a");
+            FeedItem item = DBReader.getFeedItem(queue.get(0).getId());
+            FeedMedia media = new FeedMedia(item, "http://abc.test/acme.mp3", 12345, "audio/mp3");
+            item.setMedia(media);
+            DBWriter.setFeedMedia(media).get();
+            DBWriter.setFeedItem(item).get();
+        }
+
+        PlaybackServiceTaskManager pstm = new PlaybackServiceTaskManager(c, defaultPSTM);
+        final FeedItem testItem = pstm.getQueue().get(0);
+        assertThat("The item is not yet downloaded",
+                testItem.getMedia().isDownloaded(), is(false));
+
+        { // simulate download complete (in DownloadService.MediaHandlerThread)
+            FeedItem item = DBReader.getFeedItem(testItem.getId());
+            item.getMedia().setDownloaded(true);
+            item.getMedia().setFile_url("file://123");
+            item.setAutoDownload(false);
+            DBWriter.setFeedMedia(item.getMedia()).get();
+            DBWriter.setFeedItem(item).get();
+        }
+
+        // an approximation to ensure the item update event has been posted and processed.
+        Thread.sleep(10);
```

Comment:
I don't like that, to be honst. What about using Awaitility?

Reply:
I mainly want to avoid the additional EventBus setup codes that would litter with the main test logic. The latest commits awaits for the proper `FeedItemEvent` and tries to reduce the EventBus setup's "pollution" to the main test logic.